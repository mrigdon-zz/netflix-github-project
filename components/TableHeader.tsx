import styles from "../styles/TableHeader.module.css";
import classNames from "../utils/classNames";

export interface TableHeaderProps {
  id: string;
  label: string;
  sorted: boolean;
  ascending: boolean;
  onClick(id: string): void;
}

export default function TableHeader({
  id,
  label,
  sorted,
  ascending,
  onClick,
}: TableHeaderProps) {
  return (
    <th className={styles.header}>
      <div role="button" className={styles.button} onClick={() => onClick(id)}>
        {label}{" "}
        <span
          className={classNames(
            styles.arrows,
            sorted && styles.arrowsVisible,
            ascending ? styles.arrowsAsc : styles.arrowsDesc
          )}
        >
          <span className={classNames(styles.arrow, styles.arrowUp)}>▴</span>
          <span className={classNames(styles.arrow, styles.arrowDown)}>▴</span>
        </span>
      </div>
    </th>
  );
}
