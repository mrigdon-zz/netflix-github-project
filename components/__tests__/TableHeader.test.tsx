import React from "react";
import { create } from "react-test-renderer";
import TableHeader from "../TableHeader";

describe("TableHeader", () => {
  test("unsorted", () => {
    const output = create(
      <TableHeader
        id="test"
        label="Test"
        sorted={false}
        ascending={false}
        unsortable={false}
        onClick={jest.fn()}
      />
    );
    expect(output).toMatchSnapshot();
  });

  test("sorted", () => {
    const output = create(
      <TableHeader
        id="test"
        label="Test"
        sorted={true}
        ascending={false}
        unsortable={false}
        onClick={jest.fn()}
      />
    );
    expect(output).toMatchSnapshot();
  });

  test("ascending", () => {
    const output = create(
      <TableHeader
        id="test"
        label="Test"
        sorted={true}
        ascending={true}
        unsortable={false}
        onClick={jest.fn()}
      />
    );
    expect(output).toMatchSnapshot();
  });

  test("unsortable", () => {
    const output = create(
      <TableHeader
        id="test"
        label="Test"
        sorted={false}
        ascending={false}
        unsortable={true}
        onClick={jest.fn()}
      />
    );
    expect(output).toMatchSnapshot();
  });
});
