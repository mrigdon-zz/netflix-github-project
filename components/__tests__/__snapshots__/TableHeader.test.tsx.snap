// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHeader ascending 1`] = `
<th
  className="header"
>
  <div
    className="button"
    onClick={[Function]}
    role="button"
  >
    Test
     
    <span
      className="arrows arrowsVisible arrowsAsc"
    >
      <span
        className="arrow arrowUp"
      >
        ▴
      </span>
      <span
        className="arrow arrowDown"
      >
        ▴
      </span>
    </span>
  </div>
</th>
`;

exports[`TableHeader sorted 1`] = `
<th
  className="header"
>
  <div
    className="button"
    onClick={[Function]}
    role="button"
  >
    Test
     
    <span
      className="arrows arrowsVisible arrowsDesc"
    >
      <span
        className="arrow arrowUp"
      >
        ▴
      </span>
      <span
        className="arrow arrowDown"
      >
        ▴
      </span>
    </span>
  </div>
</th>
`;

exports[`TableHeader unsortable 1`] = `
<th
  className="header"
>
  <div
    className="label"
  >
    Test
  </div>
</th>
`;

exports[`TableHeader unsorted 1`] = `
<th
  className="header"
>
  <div
    className="button"
    onClick={[Function]}
    role="button"
  >
    Test
     
    <span
      className="arrows arrowsDesc"
    >
      <span
        className="arrow arrowUp"
      >
        ▴
      </span>
      <span
        className="arrow arrowDown"
      >
        ▴
      </span>
    </span>
  </div>
</th>
`;
